### DISCUSSION TITLE
Couldn't find executable for firefox browser using default runner (devtools:puppeteer)

### DISCUSSION THREAD

**Discussion by batallo** [webdriverio/webdriverio#7916] [Q&A] [answered]
I'm using webdriverIO for some automation testing and have recently migrated from 'selenium-standalone' service to default wdio devtools protocol.

- wdio v.7.16.12
- firefox v.95.0.2

From that point I can't start testing with firefox browser:

```
INFO @wdio/cli:launcher: Run onPrepare hook
INFO @wdio/cli:launcher: Run onWorkerStart hook
INFO @wdio/local-runner: Start worker 0-0 with arg: run,wdio.conf.js
INFO @wdio/local-runner: Run worker command: run
...
INFO devtools:puppeteer: Initiate new session using the DevTools protocol
ERROR @wdio/runner: Error: Couldn't find executable for browser
...
INFO @wdio/cli:launcher: Run onComplete hook
```

I've tried different combinations of options with 'wdio:devtoolsOptions' and 'moz:firefoxOptions'.

Plus checked whether could help dumpio: true, and 'moz:debuggerAddress': true options.

Also I've tried substitution browserName with product and adding binary and executablePath to capabilities.

When passing binary: 'path.to.firefox' to 'moz:firefoxOptions' options, the error message changes to:
```
ERROR @wdio/runner: Error: Only Nightly release channel is supported in Devtools/Puppeteer for Firefox. Refer to the following issue:
...
```
But as far as I understand, starting from ffox 86.0, there is no need to use specifically Nighly versions.

Any ideas how it could be fixed in webdriverIO (without installing separately puppeteer or puppeteer-firefox)?

Thanks!

**Comment by christian-bromann** [comment, marked as answer]
> But as far as I understand, starting from ffox 86.0, there is no need to use specifically Nighly versions.

Do you have a changelog reference or anything that would state this?

**Reply by batallo** [reply]
Just encountered that idea e.g. [here](https://github.com/puppeteer/puppeteer/issues/6894) and in the first FAQ [here](https://wiki.mozilla.org/WebDriver/RemoteProtocol). Hope I haven't took it completely wrong.

The only changelog where firefox is mentioned is [this one](https://github.com/webdriverio/webdriverio/pull/7027), but there's nothing about using/not using Nightly version

**Reply by christian-bromann** [reply]
These references don't indicate that CDP has landed in Firefox stable yet. As long as this doesn't happen, we need to use Nightly.

**Reply by batallo** [reply]
@christian-bromann thanks for your answer! I've eventually installed Firefox Nightly and tests are running on it

**Reply by MeghaMShettar** [reply]
I installed Firefox Nightly , it started working , Thanks
